workitem/workitemtype_repository_bench_test.go

```diff
@@ -0,0 +1,161 @@
+package workitem_test
+
+import (
+	"testing"
+
+	"golang.org/x/net/context"
+
+	"github.com/almighty/almighty-core/gormsupport/cleaner"
+	gormbench "github.com/almighty/almighty-core/gormtestsupport/benchmark"
+	"github.com/almighty/almighty-core/migration"
+	"github.com/almighty/almighty-core/path"
+	"github.com/almighty/almighty-core/space"
+	testsupport "github.com/almighty/almighty-core/test"
+	"github.com/almighty/almighty-core/workitem"
+)
+
+type BenchWorkItemTypeRepository struct {
+	gormbench.DBBenchSuite
+	clean func()
+	repo  workitem.WorkItemTypeRepository
+	ctx   context.Context
+}
+
+func BenchmarkRunWorkItemTypeRepository(b *testing.B) {
+	testsupport.Run(b, &BenchWorkItemTypeRepository{DBBenchSuite: gormbench.NewDBBenchSuite("../config.yaml")})
+}
+
+func (s *BenchWorkItemTypeRepository) SetupSuite() {
+	s.DBBenchSuite.SetupSuite()
+	s.ctx = migration.NewMigrationContext(context.Background())
+	s.DBBenchSuite.PopulateDBBenchSuite(s.ctx)
+}
+
+func (s *BenchWorkItemTypeRepository) SetupBenchmark() {
+	s.clean = cleaner.DeleteCreatedEntities(s.DB)
+	s.repo = workitem.NewWorkItemTypeRepository(s.DB)
+}
+
+func (s *BenchWorkItemTypeRepository) TearDownBenchmark() {
+	s.clean()
+}
+
+func (r *BenchWorkItemTypeRepository) BenchmarkLoad() {
+	r.B().ResetTimer()
+	for n := 0; n < r.B().N; n++ {
+		res := workitem.WorkItemType{}
+		db := r.DB.Model(&res).Where("id=? AND space_id=?", workitem.SystemExperience, space.SystemSpace).First(&res)
+		if db.RecordNotFound() {
+			r.B().Fail()
+		}
+		if err := db.Error; err != nil {
+			r.B().Fail()
+		}
+	}
+}
+
+func (r *BenchWorkItemTypeRepository) BenchmarkLoadTypeFromDB() {
+	r.B().ResetTimer()
+	for n := 0; n < r.B().N; n++ {
+		res := workitem.WorkItemType{}
+		db := r.DB.Model(&res).Where("id=?", workitem.SystemExperience).First(&res)
+		if db.RecordNotFound() {
+			r.B().Fail()
+		}
+		if err := db.Error; err != nil {
+			r.B().Fail()
+		}
+	}
+}
+
+func (r *BenchWorkItemTypeRepository) BenchmarkListPlannerItems() {
+	r.B().ResetTimer()
+	for n := 0; n < r.B().N; n++ {
+		var rows []workitem.WorkItemType
+		path := path.Path{}
+		db := r.DB.Select("id").Where("space_id = ? AND path::text LIKE '"+path.ConvertToLtree(workitem.SystemPlannerItem)+".%'", space.SystemSpace.String())
```

Comment:
instead of doing a raw scan here, can't we rely on the workitem repository code and measure/bench the time it takes to convert the JSON fields into a `workitem.WorkItem` struct ?